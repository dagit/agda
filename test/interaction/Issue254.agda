
module Issue254 where

data Unit : Set where
  * : Unit

data Nat : Set where
  zero : Nat
  suc  : Nat → Nat

data Vec : Nat → Set where
  cons : ∀ n → Vec (suc n)

remove : ∀ n → Nat → Vec (suc n) → Unit
remove n x (cons .n) with *
remove n x (cons .n) | * = {!!}
