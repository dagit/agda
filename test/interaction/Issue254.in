top_command (cmd_load currentFile [])

-- Case split
goal_command 0 cmd_make_case "n"
