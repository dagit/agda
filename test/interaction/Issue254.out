(agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-info-action "*Type-checking*" "Checking Issue254 (Issue254.agda).\n" t)
(agda2-info-action "*Type-checking*" "Finished Issue254.\n" t)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : Unit\n" nil)
((last . 1) . (agda2-goals-action '(0)))
((last . 2) . (agda2-make-case-action '("remove zero x (cons .zero) | * = ?" "remove (suc y) x (cons .(suc y)) | * = ?")))
((last . 1) . (agda2-goals-action '(0)))
